hexrdgui/median_filter_dialog.py

```diff
@@ -0,0 +1,40 @@
+from hexrdgui.hexrd_config import HexrdConfig
+from hexrdgui.ui_loader import UiLoader
+
+
+class MedianFilterDialog:
+
+    def __init__(self, parent=None):
+        loader = UiLoader()
+        self.ui = loader.load_file(
+            'median_filter_intensity_correction_dialog.ui',
+            parent
+        )
+
+        self.ui.kernel_size.setValue(self.kernel_size)
+
+        self.update_gui()
+        self.setup_connections()
+
+    def update_gui(self):
+        # Cannot exceed array dimensions
+        img = HexrdConfig().image(HexrdConfig().detector_names[0], 0)
+        upper_bound = min(img.shape[0], img.shape[1])
+        self.ui.kernel_size.setMaximum(upper_bound)
```

Comment:
Do we need to check the shape of all images? Or just the image for the first detector?

I could imagine this filter being used for other instrument types too.

Reply:
Actually, this should either be a per-image setting or use the smallest height/width as the max to prevent the kernel size from exceeding the shape of any of the images...

Also this value is a scalar based on this exact use case, but it could also be [a list of two](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.medfilt2d.html#scipy.signal.medfilt2d).

Maybe we should just use the overall minimum for now and leave a comment as a reminder on how this can be extended in the future if there is a use for it?

Reply:
Sure! Let's just use the overall minimum for now.

Maybe you can change it to something like this?

```python
upper_bound = np.iinfo(np.int32).max
for det_key in HexrdConfig().detector_names:
    img = HexrdConfig().image(det_key, 0)
    upper_bound = min(upper_bound, *img.shape)

Reply:
I had something a [little different](https://github.com/HEXRD/hexrdgui/pull/1761/commits/1c94d553f0b925306aaaa49b8577bd6c05ebaefd#diff-6f2c5e63a9f95f46c3c62e41aa54137dee64b0ee2e3866ce446ce1f78c0835d3L19-L23) stashed in case you were okay with the overall min approach - let me know if you'd prefer I use your suggested changes instead though!

Reply:
Looks fine to me!